### PR TITLE
Relation variable in rule 'then' must not be explicit

### PR DESCRIPTION
**Pull request by shiladitya-mukherjee**
## What is the goal of this PR?
BDD test to check relation variables used to infer new relations in then clause should be anonymous.

### CODE REVIEW

File: typeql/language/define.feature [hunk 1]
```diff
@@ -2472,6 +2472,23 @@ Feature: TypeQL Define Query
...
+      person plays parentship:son;
+      rule my-rule-label:
+          when {
+              $s isa person
+              $f isa person
+              $relation(father:$f, son:$s) isa parentship; $brother isa person;
+          } then {
+          $relation(father:$f, son:$brother) isa parentship;
+          };
+      """
```
Comment (krishnangovindraj) [inline typeql/language/define.feature:2491]: This likely fits better in rule-validation.feature

**Review by krishnangovindraj — CHANGES_REQUESTED** [review]

File: typeql/language/define.feature (hunk 1 above)
Comment (shiladitya-mukherjee) [reply typeql/language/define.feature:2491]: ok!

File: typeql/language/rule-validation.feature [hunk 2]
```diff
@@ -394,6 +394,25 @@ Feature: TypeQL Rule Validation
...
+          when {
+              $s isa person
+              $f isa person
+              $relation(father:$f, son:$s) isa parentship; $brother isa person;
+          } then {
+          $relation(father:$f, son:$brother) isa parentship;
+          };
+      """
+
+
```
Comment (krishnangovindraj) [inline typeql/language/rule-validation.feature:415]: This still seems out of place within the file, given there's a section of rules for errors concerning variables in the when and then

**Review by krishnangovindraj — APPROVED** [review]
Looks good to me.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**